lte/gateway/c/core/oai/tasks/amf/amf_app_state_converter.cpp

```diff
@@ -0,0 +1,240 @@
+  guti_m5_p->guamfi.amf_regionid = std::stoul(
+      guti_str.substr(idx, chars_to_read), &chars_to_read, HEX_BASE_VAL);
+  idx += chars_to_read;
+  chars_to_read                = 4;
+  guti_m5_p->guamfi.amf_set_id = std::stoul(
+      guti_str.substr(idx, chars_to_read), &chars_to_read, HEX_BASE_VAL);
+  idx += chars_to_read;
+  chars_to_read                 = 4;
+  guti_m5_p->guamfi.amf_pointer = std::stoul(
+      guti_str.substr(idx, chars_to_read), &chars_to_read, HEX_BASE_VAL);
+  idx += chars_to_read;
+  chars_to_read     = 8;
+  guti_m5_p->m_tmsi = std::stoul(
+      guti_str.substr(idx, chars_to_read), &chars_to_read, HEX_BASE_VAL);
+  // Logs
+  // OAILOG_DEBUG_GUTI(guti_m5_p);
+}
+
+// Converts Map<guti_m5_t,uint64_t> to proto
+void AmfNasStateConverter::map_guti_uint64_to_proto(
+    map_guti_m5_uint64_t guti_map,
+    google::protobuf::Map<std::string, uint64_t>* proto_map) {
+  std::string guti_str;
+  for (auto& elm : guti_map.umap) {
+    guti_str               = amf_app_convert_guti_m5_to_string(elm.first);
+    (*proto_map)[guti_str] = elm.second;
+  }
+}
+
+// Converts Proto to Map<uint64_t,uint64_t>
+void AmfNasStateConverter::proto_to_map_uint64_uint64(
+    const google::protobuf::Map<uint64_t, uint64_t>& proto_map,
+    map_uint64_uint64_t* map) {
+  for (auto const& kv : proto_map) {
+    uint64_t id          = kv.first;
+    uint64_t val         = kv.second;
+    magma::map_rc_t m_rc = map->insert(kv.first, kv.second);
+
+    if (m_rc != magma::MAP_OK) {
+      OAILOG_ERROR(
+          LOG_UTIL, "Failed to insert value %lu in table %s: error: %s\n", val,
+          map->name.c_str(), map_rc_code2string(m_rc).c_str());
+    }
+  }
+}
+
+// Converts Proto to Map<guti_m5_t,uint64_t>
+void AmfNasStateConverter::proto_to_guti_map(
+    const google::protobuf::Map<std::string, uint64_t>& proto_map,
+    map_guti_m5_uint64_t* guti_map) {
+  for (auto const& kv : proto_map) {
+    amf_ue_ngap_id_t amf_ue_ngap_id = kv.second;
+    std::unique_ptr<guti_m5_t> guti = std::make_unique<guti_m5_t>();
+    memset(guti.get(), 0, sizeof(guti_m5_t));
+    // Converts guti to string.
+    amf_app_convert_string_to_guti_m5(guti.get(), kv.first);
+
+    guti_m5_t guti_received = *guti.get();
+    magma::map_rc_t m_rc    = guti_map->insert(guti_received, amf_ue_ngap_id);
+    if (m_rc != magma::MAP_OK) {
+      OAILOG_ERROR(
+          LOG_AMF_APP,
+          "Failed to insert amf_ue_ngap_id %lu in GUTI table, error: %s\n",
+          amf_ue_ngap_id, map_rc_code2string(m_rc).c_str());
+    }
+  }
+}
+
+/*********************************************************
+ *                AMF app state<-> Proto                  *
+ * Functions to serialize/desearialize AMF app state      *
+ * The caller is responsible for all memory management    *
+ **********************************************************/
+
+void AmfNasStateConverter::state_to_proto(
+    const amf_app_desc_t* amf_nas_state_p,
+    magma::lte::oai::MmeNasState* state_proto) {
```

Comment:
add `using magma::lte::oai::MmeNasState` on top for commonly used objects

Reply:
Comment Addressed